packages/syft/src/syft/node/routes.py

```diff
@@ -52,37 +54,64 @@ def make_routes(worker: Worker) -> APIRouter:
     async def get_body(request: Request) -> bytes:
         return await request.body()
 
-    def _blob_url(peer_uid: UID, presigned_url: str) -> str:
+    def _get_node_connection(peer_uid: UID) -> NodeConnection:
         # relative
         from ..service.network.node_peer import route_to_connection
 
         network_service = worker.get_service("NetworkService")
         peer = network_service.stash.get_by_uid(worker.verify_key, peer_uid).ok()
         peer_node_route = peer.pick_highest_priority_route()
         connection = route_to_connection(route=peer_node_route)
-        url = connection.to_blob_route(presigned_url)
-
-        return str(url)
+        return connection
 
     @router.get("/stream/{peer_uid}/{url_path}/", name="stream")
-    async def stream(peer_uid: str, url_path: str) -> StreamingResponse:
+    async def stream_download(peer_uid: str, url_path: str) -> StreamingResponse:
         try:
             url_path_parsed = base64.urlsafe_b64decode(url_path.encode()).decode()
         except binascii.Error:
             raise HTTPException(404, "Invalid `url_path`.")
 
         peer_uid_parsed = UID.from_string(peer_uid)
 
-        url = _blob_url(peer_uid=peer_uid_parsed, presigned_url=url_path_parsed)
-
         try:
-            resp = requests.get(url=url, stream=True)  # nosec
-            resp.raise_for_status()
+            peer_connection = _get_node_connection(peer_uid_parsed)
+            url = peer_connection.to_blob_route(url_path_parsed)
+            stream_response = peer_connection._make_get(url.path, stream=True)
         except requests.RequestException:
             raise HTTPException(404, "Failed to retrieve data from domain.")
 
-        return StreamingResponse(
-            resp.iter_content(chunk_size=None), media_type="text/event-stream"
+        return StreamingResponse(stream_response, media_type="text/event-stream")
+
+    async def read_request_body_in_chunks(
+        request: Request,
+    ) -> AsyncGenerator[bytes, None]:
+        async for chunk in request.stream():
+            yield chunk
+
+    @router.put("/stream/{peer_uid}/{url_path}/", name="stream")
+    async def stream_upload(peer_uid: str, url_path: str, request: Request) -> Response:
```

Comment:
API endpoint to stream download Blob url from the domain node to the end user via the gateway node.